config/v1/types_oauth.go

```diff
@@ -0,0 +1,36 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// OAuth holds cluster-wide information about OAuth.  The canonical name is `cluster`
+type OAuth struct {
```

Comment:
Possible this should be part of an "Authentication" object instead of separate.  I don't want too deep/complex objects, but 3-7 feels like a nice total number to avoid user fatigue.

Reply:
> Possible this should be part of an "Authentication" object instead of separate. I don't want too deep/complex objects, but 3-7 feels like a nice total number to avoid user fatigue.

Authentication configuration is distinct from the configuration of the oauth server.  OAuth and IDP may be worth collapsing, but not with Authentication too.